File: ES01-jakarta-kotlin/src/main/java/it/distributedsystems/model/ejb/HelloBean.java
package it.distributedsystems.model.ejb;

import jakarta.ejb.CreateException;
import jakarta.ejb.SessionBean;
import jakarta.ejb.SessionContext;

public class HelloBean implements SessionBean
{
    public String helloWorld ()   {
        return "Hello from myEjb.HelloBean";
    }

    public void ejbCreate () throws CreateException {}
    public void ejbRemove () {}
    public void setSessionContext (SessionContext ctx) {}
    public void ejbActivate () {}
    public void ejbPassivate () {}
}

File: Ex1/src/main/kotlin/it/distributedsystems/model/ejb/HelloBean.kt
package it.distributedsystems.model.ejb

import javax.ejb.CreateException
import javax.ejb.SessionBean
import javax.ejb.SessionContext
import kotlin.jvm.Throws

class HelloBean : SessionBean {

    fun helloWorld() : String {
        return "Hello from myEjb.HelloBean"
    }

    @Throws(CreateException::class)
    fun ejbCreate() {}

    override fun setSessionContext(ctx: SessionContext?) {}

    override fun ejbRemove() {}

    override fun ejbActivate() {}

    override fun ejbPassivate() {}
}